tensorpack/dataflow/imgaug/geometry.py

```diff
@@ -43,17 +72,21 @@ class RotationAndCropValid(ImageAugmentor):
         Note that this will produce images of different shapes.
     """
 
-    def __init__(self, max_deg, interp=cv2.INTER_LINEAR):
+    def __init__(self, max_deg, interp=cv2.INTER_LINEAR, step_deg=0):
```

Comment:
two reasons that relate to the fact that the integer division in not symmetric for positive and negative angles:

1. if max_deg<180 and using step_deg, you could get angles with negative value > max_deg, which breaks user expectation.
2. if max_deg<180 you dont get equal probabilities. say mag_deg=90 and step_deg=90, you get only -90 and 0 angles, missing out on +90